pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -330,8 +330,32 @@ public long getLatestIngestionTimestamp() {
    */
   private boolean isNoDictionaryColumn(Set<String> noDictionaryColumns, Set<String> invertedIndexColumns,
       Set<String> textIndexColumns, FieldSpec fieldSpec, String column) {
-    return textIndexColumns.contains(column) || (noDictionaryColumns.contains(column) && fieldSpec.isSingleValueField()
-        && !invertedIndexColumns.contains(column));
+    if (textIndexColumns.contains(column)) {
```

Comment:
The description of textIndexColumns is missing in this method.

Reply:
done

Reply:
This check here seems a little dangerous to me. We do have column level settings, it is better to throw an exception if a column has both text index as well as dictionary.

Reply:
Actually it is not needed anymore. I already do the validation upfront in TableConfig (already in master). So we can remove it. Per column config is enough

Reply:
Excellent. I think we still rename the method as something along the lines if `shouldCreateDictionaryForColumn()`, since it has some logic and is not just checking the table config for noDictionary setting,